CRM/Member/Form/MembershipRenewal.php

```diff
@@ -598,6 +598,10 @@ public function postProcess() {
 
       $payment = CRM_Core_Payment::singleton($this->_mode, $this->_paymentProcessor, $this);
 
+      if ($paymentParams['auto_renew']) {
+        $contributionRecurParams = $this->processRecurringContribution($paymentParams);
```

Comment:
I think it's OK. The recurring contribution is created before the payment is attempted.

One of the things we discussed quite a bit in Denver was creating the pending records for any payment being made & leaving them in pending state if the payment didn't happen then  - or failed. (CRM-16417 is about this). The idea being 

a) people want to know what transactions were attempted - even if they fail - so they can follow up. We agreed this in principle with dgg eons ago but have only very recently started to change the code to this

b) it simplifies the code if we ALWAYS have a two step process of creating a pending order & later completing it (the timing for completion might vary but the code called shouldn't).

I've been working to move 4.7 to this 2 step. 4.6 obviously not so much - but I think it sonewhat justifies that doing this in this order is 'part of the direction we are going in'